### PR TITLE
Add option to change the volume of music tracks

### LINKED ISSUE #430: MUTE music

### PR DESCRIPTION
**Pull request by dyceron**
If someone wants to change the music volume for whatever reason.

Fixes #430

### CODE REVIEW

File: src/open_samus_returns_rando/specific_patches/cosmetic_patches.py [hunk 1]
```diff
@@ -48,3 +49,14 @@ def music_shuffle(editor: PatcherEditor, configuration: dict) -> None:
         original_track = f"sounds/streams/music/{original}.bcwav"
         new_track = temp_dict[f"sounds/streams/music/{new}.bcwav"]
         editor.replace_asset(original_track, new_track)
+
+
+def volume_patches(editor: PatcherEditor, configuration: dict) -> None:
+    sounds = editor.get_file("system/snd/scenariomusicdefs.bmdefs", Bmdefs)
+    properties = sounds.raw["sounds"]
+    for sound in sounds:
```
Comment (Miepee) [inline src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:57]: I think this was supposed to be "for sound in properties"?

Comment (Miepee) [reply src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:57]: Also if the sound variable is never used, i'd personally either indicate it by renaming it to _sounds or just _

**Comment by dyceron** [comment]
Also supports changing the volume of environmental_sfx

File: src/open_samus_returns_rando/files/schema.json [hunk 2]
```diff
@@ -523,6 +523,24 @@
                         }
                     },
                 "default": {}
+                },
+                "volume_adjustments": {
+                    "description": "Applies a multiplier to all music tracks to adjust their volumes.",
+                    "type": "object",
+                    "properties": {
+                        "music": {
```
Comment (Miepee) [inline src/open_samus_returns_rando/files/schema.json:531]: please indicate that this is a float via the `number` type.

File: src/open_samus_returns_rando/files/schema.json [hunk 3]
```diff
@@ -523,6 +523,24 @@
...
+                            "minimum": 0,
+                            "maximum": 1
+                        },
+                        "environment_sfx": {
```
Comment (Miepee) [inline src/open_samus_returns_rando/files/schema.json:535]: dittor

File: src/open_samus_returns_rando/specific_patches/cosmetic_patches.py [hunk 4]
```diff
@@ -48,3 +50,37 @@ def music_shuffle(editor: PatcherEditor, configuration: dict) -> None:
         original_track = f"sounds/streams/music/{original}.bcwav"
         new_track = temp_dict[f"sounds/streams/music/{new}.bcwav"]
         editor.replace_asset(original_track, new_track)
+
+
+def volume_patches(editor: PatcherEditor, configuration: dict) -> None:
+    music = configuration["volume_adjustments"]["music"]
+    environment_sfx = configuration["volume_adjustments"]["environment_sfx"]
+    # If there are no custom values, quit
+    if music == 1 and environment_sfx == 1:
```
Comment (Miepee) [inline src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:59]: I'd do this check slightly differently:
have `_adjust_music_volume()` and `_adjust_sfx_volume()` as seperate functios
Then `if music != 1: _adjust_music_volume(); if environment_sfx != 1: _adjust_sfx_volume();`
That way, you also skip out on unnecessay modification if you do `music = 1, environment_sfx = 0`

File: src/open_samus_returns_rando/specific_patches/cosmetic_patches.py [hunk 5]
```diff
@@ -48,3 +50,37 @@ def music_shuffle(editor: PatcherEditor, configuration: dict) -> None:
...
+        sound["volume"] *= music
+
+    enemies_list = sound_defs.raw["enemies_list"]
+    for enemy in enemies_list:
+        areas = enemy["areas"]
+        for area in areas:
+            layers = area["layers"]
+            for layer in layers:
+                states = layer["states"]
+                for state in states:
```
Comment (Miepee) [inline src/open_samus_returns_rando/specific_patches/cosmetic_patches.py:76]: why not just do this
```suggestion
    for enemy in enemies_list:
        for area in enemy["areas"]:
            for layer in area["layers"]:
                for state in layer["states"]:
```
You're never using the other variables anyway.

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**